### PR TITLE
 Garbage Collection and memory snapshot code for debugging partitioning pipeline (target as master branch)

### PR DESCRIPTION
**Pull request by kylasa**
commit e605a550b3783dd5f24eb39b6873a2e0e79be9c7
Author: kylasa <kylasa@gmail.com>
Date:   Thu Sep 15 14:45:39 2022 -0700

    Delete pyproject.toml

commit f2db9e700d817212b67b5227f6472d218f0c74f2
Author: kylasa <kylasa@gmail.com>
Date:   Thu Sep 15 14:44:40 2022 -0700

    Changes suggested by isort program to sort imports.

commit 5a6078beac6218a4f1fb378c169f04dda7396425
Author: kylasa <kylasa@gmail.com>
Date:   Thu Sep 15 14:39:50 2022 -0700

    addressing code review comments from the CI process.

commit c8e92decb7aebeb32c7467108e16f058491443ab
Author: kylasa <kylasa@gmail.com>
Date:   Wed Sep 14 18:23:59 2022 -0700

    Corrected a typo in the import statement

commit 14ddb0e9b553d5be3ed2c50d82dee671e84ad8c9
Author: kylasa <kylasa@gmail.com>
Date:   Tue Sep 13 18:47:34 2022 -0700

    Memory snapshot code for debugging memory footprint of the graph partitioning pipeline

Squashed commit done

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tools/dispatch_data.py [hunk 1]
```diff
@@ -69,6 +70,7 @@ def main():
     parser.add_argument('--out-dir', type=str, help='Location of the output directory where the graph partitions will be created by this pipeline')
     parser.add_argument('--ip-config', type=str, help='File location of IP configuration for server processes')
     parser.add_argument('--master-port', type=int, default=12345, help='port used by gloo group to create randezvous point')
+    parser.add_argument('--log', type=str, default="info", help='to enable logging at a given level')
```
Comment (Rhett-Ying) [inline tools/dispatch_data.py:73]: `--log_level` is better, I think.

Comment (Rhett-Ying) [reply tools/dispatch_data.py:73]: and could we print out all supported levels in the help message?

**Review by Rhett-Ying — COMMENTED** [review]
could we utilize https://github.com/dmlc/dgl/blob/166b273becae0438d0114daf5346b80ebf1a4333/python/dgl/partition.py#L235 to record the peak memory used by each process/rank. I think this info is really helpful as we know the larget RAM we should prepare before partition.

File: tools/distpartitioning/convert_partition.py [hunk 2]
```diff
@@ -117,8 +123,19 @@ def create_dgl_object(graph_name, num_parts, \
     node_map_val = {ntype: [] for ntype in ntypes}
     edge_map_val = {etype.split(":")[1]: [] for etype in etypes}
 
-    shuffle_global_nids, ntype_ids, global_type_nid = node_data[constants.SHUFFLE_GLOBAL_NID], \
-            node_data[constants.NTYPE_ID], node_data[constants.GLOBAL_TYPE_NID]
+    memory_snapshot("CreateDglObj_AssignNodeData", part_id)
```
[the remark this answers is not in the document]
Comment (kylasa) [reply tools/distpartitioning/convert_partition.py:126]: Now I changed all such instances to read 'CreateDGLObj_'

File: tools/dispatch_data.py (hunk 1 above)
Comment (kylasa) [reply tools/dispatch_data.py:73]: Done as suggested.

**Review by Rhett-Ying — APPROVED** [review]

**Comment by jermainewang** [comment]
@kylasa Can you list the memory consumption before/after this PR? Want to understand how effective is this.

File: tools/distpartitioning/convert_partition.py [hunk 3]
```diff
@@ -117,8 +123,19 @@ def create_dgl_object(graph_name, num_parts, \
...
+    gc.collect()
+
+    ntype_ids = node_data[constants.NTYPE_ID]
+    node_data.pop(constants.NTYPE_ID)
+    gc.collect()
+
+    global_type_nid = node_data[constants.GLOBAL_TYPE_NID]
+    node_data.pop(constants.GLOBAL_TYPE_NID)
+    node_data = None
+    gc.collect()
```
Comment (zheng-da) [inline tools/distpartitioning/convert_partition.py:138]: you only need to call gc once here. you pop data from the map but the data is still referenced by the variable. then when you call gc, it cannot collect memory. only when you dereference `node_data`, you should trigger GC.

File: tools/distpartitioning/convert_partition.py [hunk 4]
```diff
@@ -133,11 +150,32 @@ def create_dgl_object(graph_name, num_parts, \
...
+
+    global_edge_id = edge_data[constants.GLOBAL_TYPE_EID]
+    edge_data.pop(constants.GLOBAL_TYPE_EID)
+    gc.collect()
+
+    etype_ids = edge_data[constants.ETYPE_ID]
+    edge_data.pop(constants.ETYPE_ID)
+    edge_data = None
+    gc.collect()
```
Comment (zheng-da) [inline tools/distpartitioning/convert_partition.py:177]: similar comments here

**Pull request merged**